StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1108BlockStatementsMustNotContainEmbeddedComments.cs

```diff
@@ -58,19 +61,90 @@ public class SA1108BlockStatementsMustNotContainEmbeddedComments : DiagnosticAna
         private static readonly ImmutableArray<DiagnosticDescriptor> SupportedDiagnosticsValue =
             ImmutableArray.Create(Descriptor);
 
+        private readonly SyntaxKind[] supportedKinds = new[]
+        {
+            SyntaxKind.ForEachStatement,
+            SyntaxKind.ForStatement,
+            SyntaxKind.WhileStatement,
+            SyntaxKind.DoStatement,
+            SyntaxKind.IfStatement,
+            SyntaxKind.ElseClause,
+            SyntaxKind.LockStatement,
+            SyntaxKind.TryStatement,
+            SyntaxKind.CatchClause,
+            SyntaxKind.FinallyClause,
+            SyntaxKind.CheckedStatement,
+            SyntaxKind.UncheckedStatement,
+            SyntaxKind.FixedStatement
+        };
+
+        /// <inheritdoc/>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => SupportedDiagnosticsValue;
+
         /// <inheritdoc/>
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(this.AnalyzeBlock, SyntaxKind.Block);
+            context.RegisterSyntaxNodeAction(this.AnalyzeSwitch, SyntaxKind.SwitchStatement);
+        }
+
+        private void AnalyzeSwitch(SyntaxNodeAnalysisContext context)
         {
-            get
+            var switchStatement = (SwitchStatementSyntax) context.Node;
+            var openBraceToken = switchStatement.OpenBraceToken;
+            if (openBraceToken.IsMissing)
             {
-                return SupportedDiagnosticsValue;
+                return;
             }
+
+            var previousToken = openBraceToken.GetPreviousToken();
+            if (previousToken.IsMissing)
```

Comment:
I thought about some incorrect code, but I checked it again and, you're right, it's a useless if statement.